plenum/common/test_network_setup.py

```diff
@@ -46,142 +46,102 @@ def writeNodeParamsFile(filePath, name, nPort, cPort):
+        domainLedger = cls.init_domain_ledger(appendToLedgers, baseDir, config,
+                                              envName, domainTxnFieldOrder)
 
-        domainLedger = Ledger(CompactMerkleTree(),
-                              serializer=CompactSerializer(fields=
-                                                           domainTxnFieldOrder),
-                              dataDir=baseDir,
-                              fileName=domainTxnFile)
+        trustee_txn = Member.nym_txn(trustee_def.nym, trustee_def.name, 'TRUSTEE')
+        domainLedger.add(trustee_txn)
 
-        if not appendToLedgers:
-            poolLedger.reset()
-            domainLedger.reset()
-
-        trusteeName = "Trustee1"
-        sigseed = TestNetworkSetup.getSigningSeed(trusteeName)
-        verkey = Signer(sigseed).verhex
-        trusteeNym = TestNetworkSetup.getNymFromVerkey(verkey)
-        txn = {
-            TARGET_NYM: trusteeNym,
-            TXN_TYPE: NYM,
-            # TODO: Trustees dont exist in Plenum, but only in Sovrin.
-            # This should be moved to Sovrin
-            ROLE: TRUSTEE,
-            ALIAS: trusteeName,
-            TXN_ID: sha256(trusteeName.encode()).hexdigest()
-        }
-        domainLedger.add(txn)
-
-        steward1Nym = None
-        for num in range(1, nodeCount + 1):
-            stewardName = "Steward" + str(num)
-            sigseed = TestNetworkSetup.getSigningSeed(stewardName)
-            verkey = Signer(sigseed).verhex
-            stewardNym = TestNetworkSetup.getNymFromVerkey(verkey)
-            txn = {
-                TARGET_NYM: stewardNym,
-                TXN_TYPE: NYM,
-                ROLE: STEWARD,
-                ALIAS: stewardName,
-                TXN_ID: sha256(stewardName.encode()).hexdigest()
-            }
-            if num == 1:
-                steward1Nym = stewardNym
-            else:
-                # The first steward adds every steward
-                txn[f.IDENTIFIER.nm] = steward1Nym
-            domainLedger.add(txn)
+        for sd in steward_defs:
+            nym_txn = Member.nym_txn(sd.nym, sd.name, role=STEWARD,
+                                     creator=trustee_def.nym)
+            domainLedger.add(nym_txn)
 
-            nodeName = "Node" + str(num)
-            nodePort, clientPort = startingPort + (num * 2 - 1), startingPort \
-                                   + (num * 2)
-            ip = ips[num - 1]
-            sigseed = TestNetworkSetup.getSigningSeed(nodeName)
-            if nodeNum == num:
-                _, verkey = initLocalKeys(nodeName, baseDir, sigseed, True,
-                                     config=config)
-                _, verkey = initLocalKeys(nodeName+CLIENT_STACK_SUFFIX, baseDir,
-                                     sigseed, True, config=config)
+        for nd in node_defs:
+
+            if nd.idx in _localNodes:
+                _, verkey = initLocalKeys(nd.name, baseDir, nd.sigseed, True, config=config)
```

Comment:
Why we init node keys only, not keys for both stacks (node and client) like before?